spec/isolator/adapters/redis_adapter_spec.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Redis do
+  before { Isolator.enable! }
+
+  subject(:client) { Redis.new }
+
+  specify do
+    expect { client.get('key') }.to raise_error(Isolator::RedisAccessError)
```

Comment:
That's exactly the case where we don't want to raise an exception, 'cause _reading_ is a safe operation (well, almost safe, it's still prone to network failures but they don't break atomicity).

That's why the proposal told about monitoring background jobs – specific Redis use case, and not general Redis usage.

Reply:
Yeah, you're right. I just feel like it's better to patch redis than all of gems for background jobs. Because In the end, we still need to patch redis client. I can skip raising exceptions for safe operations. What do you think about that?

Reply:
I think, these are different levels of logic; it still make sense to patch (or monitor?) bg jobs adapters (well, not adapters, but ActiveJob and Sidekiq – that would be enough for now).

Thus we can provide more meaningful errors.

Monitoring Redis is a more low-level feature.